openlibrary/catalog/add_book/__init__.py

```diff
@@ -685,8 +686,213 @@ def load_data(rec, account_key=None):
     return reply
 
 
+def add_missing_int_and_str_values_to_edition(
+    source_rec: dict,
+    edition: Edition,
+    need_edition_save: bool,
+    overwrite: bool = False,
+) -> tuple[Edition, bool]:
+    """
+    Add missing string-like and int-like values from source_rec to the target edition.
+
+    :param source_rec: A `rec` dictionary *returned from build_query()*.
+    :param edition: A the matched target edition that will be updated with any new values.
+    :param need_edition_save: True if the edition has been modified and needs saving.
+    :param overwrite: If true, use source_rec values to overwrite those in edition.
+    :return: A tuple containing the (possibly modified) edition and a bool indicating
+        whether the edition needs to be saved.
+    """
+    # Update `edition` str and int value fields with `rec_edition` values. Don't overwrite
+    # existing fields by default.
+    for field, value in source_rec.items():
+        # Don't overwrite existing fields by default.
+        if field in edition and overwrite is False:
+            continue
+
+        if isinstance(value, (str, int)):
+            edition[field] = value
+            need_edition_save = True
+            continue
+
+        if isinstance(value, dict):
+            if value.get('type') == '/type/text' or value.get('type') == '/type/int':
```

Comment:
Get the type once, not twice.
```suggestion
            if value.get('type') in ('/type/int', '/type/text'):
```